### PR TITLE
Fix: Path to package in lerna workspace

### PR DESCRIPTION
**Pull request by literat**
* package is linked in node_modules as entire package directory
  * src or dist are present
  * so we can use build path or add support for typescript in src

### CODE REVIEW

File: examples/web-react/webpack.config.js [hunk 1]
```diff
@@ -27,5 +27,8 @@ module.exports = () => ({
   resolve: {
     extensions: ['.js', '.jsx'],
     modules: ['src', 'node_modules'],
+    alias: {
+      '@lmc-eu/spirit-web-react': Path.resolve(__dirname, '../../packages/web-react/dist/_esNext'),
```
[the remark this answers is not in the document]
Comment (literat) [reply examples/web-react/webpack.config.js]: This is outdated. I have removed entire examples from the workspace and refactored the building process to match our needs.

File: packages/web-react/package.json [hunk 2]
```diff
@@ -12,9 +12,9 @@
     "url": "https://github.com/lmc-eu/spirit-design-system.git",
     "directory": "packages/web-react"
   },
-  "module": "_es2015/index.js",
```
Comment (literat) [inline packages/web-react/package.json:15]: Package structure is treated in a same way as a directory so to use these directories we must also include them in out import path or create and alias.

File: packages/web-react/scripts/prepareDist.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
+// The Spirit Web React source that is published to npm is located in the
```
Comment (literat) [inline packages/web-react/scripts/prepareDist.js:1]: This script with aid of `entryPoints.js` just add simple `package.json` to every directory to make `components` or `components/Button` accessible via import.

File: packages/web-react/scripts/resolveModuleIds.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
+import * as fs from "fs";
```
Comment (literat) [inline packages/web-react/scripts/resolveModuleIds.ts:1]: This script adds `.js` extension to export path to make it accessible via bundlers.

**Comment by literat** [comment]
No we can import our components as
```javascript
import { Button } from '@lmc-eu/spirit-web-react';
// or
import { Button } from '@lmc-eu/spirit-web-react/components';
// or
import { Button } from '@lmc-eu/spirit-web-react/components/Button';
```

**Review by adamkudrna — COMMENTED** [review]
I'm not able to review the scripts in `web-react` package, maybe ESlint or a JS developer would help better than me 🙂.

My only note is that `examples/web-react` isn't working:

```
Uncaught Error: Cannot find module '@lmc-eu/spirit-web-react/src/components/Button'
    webpackMissingModule webpack://@lmc-eu/spirit-example-web-react/./src/demo.jsx?:4
    <anonymous> webpack://@lmc-eu/spirit-example-web-react/./src/demo.jsx?:4
    jsx http://localhost:63342/spirit-design-system/examples/web-react/built/demo.js:99
    __webpack_require__ http://localhost:63342/spirit-design-system/examples/web-react/built/demo.js:133
    <anonymous> http://localhost:63342/spirit-design-system/examples/web-react/built/demo.js:156
    <anonymous> http://localhost:63342/spirit-design-system/examples/web-react/built/demo.js:158
demo.js line 99 > eval:4:50
```

Can we fix it? Or is it going to work once this is merged and released?

File: packages/web-react/scripts/resolveModuleIds.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+  isRelative(id: string) {
+    return id.startsWith("./") || id.startsWith("../");
+  }
+
+  normalizeSourceString(file: string, source?: Node | null) {
+    if (source && n.StringLiteral.check(source)) {
+      // We mostly only worry about normalizing _relative_ module identifiers,
+      // which start with a ./ or ../ and refer to other modules within the
+      // @lmc-ei/spirit-web-react package, but we also manually normalize one non-relative
```
[the remark this answers is not in the document]
Comment (literat) [reply packages/web-react/scripts/resolveModuleIds.ts]: Fixed :-)

**Comment by literat** [comment]
> I'm not able to review the scripts in `web-react` package, maybe ESlint or a JS developer would help better than me 🙂.
> 
> My only note is that `examples/web-react` isn't working:
> 
> ```
> Uncaught Error: Cannot find module '@lmc-eu/spirit-web-react/src/components/Button'
>     webpackMissingModule webpack://@lmc-eu/spirit-example-web-react/./src/demo.jsx?:4
>     <anonymous> webpack://@lmc-eu/spirit-example-web-react/./src/demo.jsx?:4
>     jsx http://localhost:63342/spirit-design-system/examples/web-react/built/demo.js:99
>     __webpack_require__ http://localhost:63342/spirit-design-system/examples/web-react/built/demo.js:133
>     <anonymous> http://localhost:63342/spirit-design-system/examples/web-react/built/demo.js:156
>     <anonymous> http://localhost:63342/spirit-design-system/examples/web-react/built/demo.js:158
> demo.js line 99 > eval:4:50
> ```
> 
> Can we fix it? Or is it going to work once this is merged and released?

Well, this will be solved with another version. But you can  test it yourself by building `web-react` by `yarn build` and then `cd ./dist`. There you should run `yarn pack` to create tar package. Now you can move to that example and install it using something like `yarn add /Users/literat/Work/lmc/spirit-design-system/packages/web-react/dist/lmc-eu-spirit-web-react-v0.2.7.tgz`. After this it will be working.

I know, it is little bit difficult but I do not know better way how to pair our examples with actual version. Maybe create and publish some type of semver prerelease. What do you think?

File: packages/web-react/package.json [hunk 6]
```diff
@@ -12,9 +12,9 @@
...
-  "esnext": "_esNext/index.js",
-  "types": "_esNext/index.d.ts",
+  "main": "./main.cjs",
+  "module": "./index.js",
+  "types": "./index.d.ts",
   "dependencies": {
     "classnames": "^2.3.1",
     "react": "^17.0.2",
```
[the remark this answers is not in the document]
Comment (literat) [reply packages/web-react/package.json:20]: I think not. This package directly depends on React and React DOM. So we follow conventions to place direct dependencies into `dependencies` field. However technically it does not matter. It matter only when developing for node and install only production dependencies when going to production. Webpack and other bundlers resolves both `dependencies` and `devDependencies`.

So we can follow conventions or move all to `devDependencies` like [React does it](https://github.com/facebook/react/blob/main/package.json). Both works.

File: packages/web-react/src/index.tsx [hunk 7]
```diff
@@ -1 +1 @@
-export { Button } from './components/Button';
+export * from './components';
```
[the remark this answers is not in the document]
Comment (literat) [reply packages/web-react/src/index.tsx:1]: I do not know exactly but I took the inspiration from https://github.com/apollographql/apollo-client/blob/main/src/index.ts. I want to care about export only on final level. On main or sublevels I only want to get everything out.

So I can import Button from root or from sublevel. If I use explicit export then I must explicitly reexport it on every level.

**Pull request merged**